src/api/environment.cc

```diff
@@ -581,16 +581,21 @@ void InitializeContextRuntime(Local<Context> context) {
   }
 }
 
-bool InitializeContextForSnapshot(Local<Context> context) {
+MaybeLocal<Context> InitializeContextForSnapshot(Local<Context> context) {
   Isolate* isolate = context->GetIsolate();
   HandleScope handle_scope(isolate);
 
   context->SetEmbedderData(ContextEmbedderIndex::kAllowWasmCodeGeneration,
                            True(isolate));
-  return InitializePrimordials(context);
+
+  if (InitializePrimordials(context).IsEmpty()) {
+    return MaybeLocal<Context>();
+  }
+
+  return context;
 }
 
-bool InitializePrimordials(Local<Context> context) {
+MaybeLocal<Object> InitializePrimordials(Local<Context> context) {
```

Comment:
It’s convention to use `Maybe<bool>` for methods that either fail with a pending exception or succeed otherwise (i.e. either return `Just(true)` or `Nothing<bool>()`)

Reply:
@addaleax done